flink-autoscaler/src/main/java/org/apache/flink/autoscaler/JobVertexScaler.java

```diff
@@ -71,21 +72,31 @@ public JobVertexScaler(AutoScalerEventHandler<KEY, Context> autoScalerEventHandl
         this.autoScalerEventHandler = autoScalerEventHandler;
     }
 
-    public int computeScaleTargetParallelism(
+    public VertexScalingResult computeScaleTargetParallelism(
             Context context,
             JobVertexID vertex,
             Collection<ShipStrategy> inputShipStrategies,
             Map<ScalingMetric, EvaluatedScalingMetric> evaluatedMetrics,
             SortedMap<Instant, ScalingSummary> history,
-            Duration restartTime) {
+            Duration restartTime,
+            double backpropagationScaleFactor) {
```

Comment:
1.0 is used in tests and for the first iteration. Later, this value is set to the minimum back propagation scale factor.

Reply:
The production code doesn't actually set this to something else than 1.0. It's other code paths which use a different factor. 

I would prefer to remove the parameter and any added tests which set it directly. This will also get rid of a lot of changed lines to add the 1.0 placeholder.